### PR TITLE
Feat/soundcloud

### PR DESCRIPTION
**Pull request by hugurp**

### CODE REVIEW

File: src/FilePlayer.vue [hunk 1]
```diff
@@ -84,6 +84,11 @@
 			},
 			handleEnded() {
 				this.$emit('ended')
+				// it seems the element emits pause at the end of a track
+				// so if we are in pause when ended is handled, we should play!
+				if(!this.isPlaying) {
```
Comment (hugurp) [inline src/FilePlayer.vue]: this bit is for when `filePlayer` will finish and go to an other provider, `soundcloudPlayer`

File: src/SoundcloudPlayer.vue [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+		}
+	}
+</script>
+
+<style>
+	.SoundcloudPlayer,
+	#SoundcloudPlayerR4 {
+		height: 100%;
+	}
+	.SoundcloudPlayer iframe {
```
Comment (oskarrough) [inline src/SoundcloudPlayer.vue]: this part can be removed, since .ProviderPlayer takes care of it.

File: src/SoundcloudPlayer.vue [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+	 */
+
+	/* 
+		 Notes: soundcloud is a bit tricky
+		 - it does not seem possible to use their js on github, since they don't give api keys no more
+		 - Their widget sends weird pause events when a track is finished
+		 and changing directly the `src` on the iframe breaks the listeners
+		 - We do not, but could, handle sets, just playing the first track for now
+	 */
+	import './libs/soundcloud.js'
```
Comment (oskarrough) [inline src/SoundcloudPlayer.vue]: Should we load through NPM instead?

**Comment by oskarrough** [comment]
**Update** now it autoplays in my Chrome browser. Dunno if it understood clicking and refreshing +20 times as giving my permission to autoplay…

**Comment by oskarrough** [comment]
Clicking on a Soundcloud track triggers "Play" in the console 3-12 times. Maybe some re-binding of events?

File: src/SoundcloudPlayer.vue (hunk 3 above)
Comment (hugurp) [reply src/SoundcloudPlayer.vue]: there does not seem to be a npm module...

**Comment by hugurp** [comment]
Is it the correct removal? https://github.com/internet4000/radio4000-player/pull/136/commits/12c84b462951abed5942a4e8db6d0bac9857925f

File: src/SoundcloudPlayer.vue (hunk 2 above)
Comment (hugurp) [reply src/SoundcloudPlayer.vue]: done https://github.com/internet4000/radio4000-player/pull/136/commits/484e67fa43203ab253f7102cff4344bd03704f1b

**Comment by hugurp** [comment]
> Cool! I tested the local server a bit in Brave with autoplay allowed:
> 
> The soundcloud tracks don't autoplay. Same if I click a soundcloud track in the list. The SC player shows the pause icon as if it is playing, but it's not. Console shows:
> 
> ```
> (4) The AudioContext was not allowed to start. It must be resumed (or created) from a user gesture event handler. 
> (5) Play
> ```
> 
> In my Firefox it does autoplay!
> 
> Volume button (the square one bottom left) works well for Soundcloud. Also setting `volume` attribute manually. For Vimeo, it seems to have no effect. Also for vimeo, "our" play button doesn't control the video?

I have the same in Firefox, because I deactivated all cookies and so on.
It is needed to activate that in your Brave browser, for all providers, Youtube, Soundcloud, Vimeo. so their cookies and data are alowed to be stored. Otherwise the player does not start.

Same for Autoplay, it needs to be allowed in localhost, and also also on radio4000, is the `autoplay` is on o nthe `<radio4000-player>` html element.

**Comment by hugurp** [comment]
I do not have any of the error you mention for Vimeo's volume

**Pull request closed without merging**